src/main/resources/assets/js/components/AppComponent.jsx

```diff
@@ -23,6 +24,43 @@ define([
         "text-warning": isDeploying
       });
 
+      var hasHealth = model.get("healthChecks") != null &&
+        model.get("healthChecks").length > 0;
+
+      var healthPieColors = [
+        "#7aff68", // healthy
+        "#ff0000", // unhealthy
+        "#8e929b", // running
+        "#FFFF00", // staged
+        "#0000ff", // over-capacity
+        "#2a2C31", // unscheduled
+      ];
+      var healthPieData = [
+        { quantity: model.get("tasksHealthy") },
+        { quantity: model.get("tasksUnhealthy") },
+        { quantity: Math.max(model.get("tasksRunning") -
+          model.get("tasksHealthy") - model.get("tasksUnhealthy"), 0) },
+        { quantity: model.get("tasksStaged") },
+      ];
+
+      // cut off after `instances` many tasks, show everything above that in blue
+      var tasksSum = 0;
+      for (var i = 0; i < healthPieData.length; i++) {
+        healthPieData[i].quantity = Math.min(
+          tasksSum + healthPieData[i].quantity, model.get("instances")
+        ) - tasksSum;
+        tasksSum += healthPieData[i].quantity;
+      }
+      var unscheduledTasks = model.get("instances") -
+        model.get("tasksRunning") - model.get("tasksStaged");
+      var suspended = model.get("instances") === 0 &&
+        model.get("tasksRunning") === 0 &&
+        model.get("tasksStaged") === 0;
+      healthPieData.push(
+        { quantity: Math.max(- unscheduledTasks, 0)},
+        { quantity: suspended ? 1 : Math.max(unscheduledTasks, 0) }
```

Comment:
And no spaces in objects: {quantity: suspended ? 1 : Math.max(unscheduledTasks, 0)}, though I'm not sure that will be needed anymore.
Also, these two lines could probably be made more readable - creating variables would help.